sphinx_needs/lsp/needs_store.py

```diff
@@ -0,0 +1,110 @@
+# --------------------------------------------------------------------------
+# Licensed under the MIT license.
+# See License.txt in the project root for further license information.
+# --------------------------------------------------------------------------
+
+import importlib.util
+import json
+import logging
+import os
+import sys
+
+from sphinx_needs.lsp.exceptions import NeedlsConfigException
+
+
+class NeedsStore:
+    """Abstraction of needs database."""
+
+    def __init__(self):
+        self.docs_per_type = {}  # key: need type, val: list of doc names (str)
+        self.needs_per_doc = {}  # key: docname; val: list of needs
+        self.types = []  # list of need types actually defined in needs.json
+        self.declared_types = {}  # types declared in conf.py: {'need directive': 'need title'}
+        self.needs = {}
+        self.needs_initialized = False
+        self.docs_root = None
+        self.conf_py_path = None
+
+    def is_setup(self) -> bool:
+        """Return True if database is ready for use."""
+
+        if not self.needs_initialized:
+            return False
+
+        return True
+
+    def set_docs_root(self, docs_root: str) -> None:
+        if not os.path.exists(docs_root):
+            raise FileNotFoundError(f"Docs root directory not found: {docs_root}")
+        self.docs_root = docs_root
+
+    def set_conf_py(self, conf_py_path: str) -> None:
```

Comment:
would be better to use `pathlib.Path` instead of `str` everywhere that it represents a filepath.